### ISSUE TITLE
Bug: GitLab CI pipeline fails unexpectedly for QDD 0.2.4

### ISSUE THREAD

**Issue by marcosdiezgarcia** [Fujitsu-UTokyo-QDD/QDD#70]
Jobs get unexpectedly and abruptly killed by the GitLab runner due to some incompatibility or misconfiguration between [QDD](https://github.com/Fujitsu-UTokyo-QDD/QDD) and the GitLab CI runner.

To reproduce the error I created a new GitLab repository for testing purposes and configured a `gitlab-ci.yml` file which does the following:
1. Creates an ubuntu 24.04 image and automatically installs some preliminary commands like git, python 3.12 and cmake >= 3.25.
2. Git clones QDD official repository.
3. Creates a Python 3.12 virtual environment and installs QDD from source (following QDD install instructions).
4. Shows successful installation of `qdd-0.2.4`.
5. Runs `python -c "import qdd"`

The output to stdout is the following:
> /usr/bin/bash: line 162:  5658 Killed                  python -c "import qdd"

However, pytket-qdd and qdd run without issue when installed locally via pytket-qdd Makefile (tested on GNU/Linux Ubuntu 22.04).

gitlab-ci.yaml
```
default:
    image: ubuntu:latest
    before_script:
        - apt-get update --yes
        - apt-get install --yes python3-pip python3-build python3-virtualenv cmake git
        - virtualenv venv
        - source venv/bin/activate
        - git clone https://github.com/Fujitsu-UTokyo-QDD/QDD.git
        - cd QDD
        - bash ./scripts/local_build_and_test_new.sh build
        - pip install .
        - cd -

stages:
    - test

job-test:
    stage: test
    script:
        - python --version
        - pip show -f qdd
        - python -c "print('This is a test')"
        - python -c "import qdd"
```

See attached GitLab runner log.

[20241022--gitlab-ci-qdd-log.txt](https://github.com/user-attachments/files/17539166/20241022--gitlab-ci-qdd-log.txt)

**Comment by marcosdiezgarcia** [comment]
Thank you @kkyusuke for your feedback. Please allow me to double-check your update. Thanks.

**Comment by marcosdiezgarcia** [comment]
Details of GitLab runner environment:
```bash
$ uname -a
Linux runner-5gkczqktm-project-8400-concurrent-0 5.15.0-1074-azure #83~20.04.1-Ubuntu SMP Fri Oct 4 21:49:59 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

```bash
$ lscpu
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               2
On-line CPU(s) list:                  0,1
Vendor ID:                            GenuineIntel
BIOS Vendor ID:                       Intel(R) Corporation
Model name:                           Intel(R) Xeon(R) Platinum 8[27](https://code.mahitahi.global.fujitsu.com/marcos.diezgarcia/test/-/jobs/489723#L27)2CL CPU @ 2.60GHz
BIOS Model name:                      Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz None CPU @ 2.6GHz

(further lscpu details)
```

**Comment by kkyusuke** [comment]
I finally found that it is out of memory error.
I have added one more commit in #71, and I believe it will fix the problem.

I plan to merge these changes this week, as a new release.

**Comment by marcosdiezgarcia** [comment]
Thank you @kkyusuke for looking into this and proposing a fix.

**Comment by marcosdiezgarcia** [comment]
Hi Kimura-san @kkyusuke . I noticed you updated QDD to 0.2.5 version.

I tried again my gitab-ci script using this new version with same gitlab runner environment configuration.

The following is extracted from the gitlab runner output:

![image](https://github.com/user-attachments/assets/9c426e69-8e12-4680-b337-3087898953df)

Now running `import qdd` does not cause the gitlab runner job to get killed. The job is completed successfully. I tried a few times running the job and it succedes for each time, so I think the bug seems solved now.

Many thanks.
